### PR TITLE
Fix PaymentSheet picker on macOS Catalyst

### PR DESCRIPTION
**Pull request by davidme-stripe**
## Summary
Pickers in PaymentSheet (including the country selector) don't work, as UITextField inputViews don't appear in Catalyst. This means clicking the picker will do nothing.

I evaluated a few different alternatives for this, and the least-bad one seems to be a UIButton with an attached menu. Ideally we'd have some sort of NSComboBox equivalent on Catalyst, but I haven't been able to find a publicly available one yet.

UIMenu is only available on macOS 11, but Catalyst wasn't very usable on 10.15 (our minimum version in the README). Looking at our analytics, we don't appear to have any users before macOS 12, so I feel comfortable using this and bumping the version in the README.

## Motivation
Fixing country selection on Catalyst.

## Testing
In PaymentSheet Example

## Changelog
Added

### CODE REVIEW

**Review by wooj-stripe — COMMENTED** [review]
Code looks fine, but is it possible to add a test or two? Perhaps snapshot tests?

**Comment by davidme-stripe** [comment]
> Code looks fine, but is it possible to add a test or two? Perhaps snapshot tests?

We actually don't have *any* Catalyst tests, which does seem like an oversight. Let me hook some up.

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Link/LinkURLGeneratorTests.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ class LinkURLGeneratorTests: XCTestCase {
...
-        XCTAssertEqual(url.absoluteString, "https://checkout.link.com/#eyJsb2dnZXJNZXRhZGF0YSI6e30sImN1c3RvbWVySW5mbyI6eyJjb3VudHJ5IjoiVVMiLCJlbWFpbCI6InRlc3RAZXhhbXBsZS5jb20ifSwicGF5bWVudE9iamVjdCI6ImxpbmtfcGF5bWVudF9tZXRob2QiLCJleHBlcmltZW50cyI6e30sInBheW1lbnRVc2VyQWdlbnQiOiJ0ZXN0IiwibG9jYWxlIjoiZW4tVVMiLCJwYXRoIjoibW9iaWxlX3BheSIsInBheW1lbnRJbmZvIjp7ImN1cnJlbmN5IjoiVVNEIiwiYW1vdW50IjoxMDB9LCJtZXJjaGFudEluZm8iOnsiY291bnRyeSI6IlVTIiwiYnVzaW5lc3NOYW1lIjoiVGVzdCB0ZXN0In0sInB1Ymxpc2hhYmxlS2V5IjoicGtfdGVzdF8xMjMiLCJmbGFncyI6e30sImludGVncmF0aW9uVHlwZSI6Im1vYmlsZSJ9")
+        XCTAssertEqual(url.absoluteString, "https://checkout.link.com/#eyJjdXN0b21lckluZm8iOnsiY291bnRyeSI6IlVTIiwiZW1haWwiOiJ0ZXN0QGV4YW1wbGUuY29tIn0sImV4cGVyaW1lbnRzIjp7fSwiZmxhZ3MiOnt9LCJpbnRlZ3JhdGlvblR5cGUiOiJtb2JpbGUiLCJsb2NhbGUiOiJlbi1VUyIsImxvZ2dlck1ldGFkYXRhIjp7fSwibWVyY2hhbnRJbmZvIjp7ImJ1c2luZXNzTmFtZSI6IlRlc3QgdGVzdCIsImNvdW50cnkiOiJVUyJ9LCJwYXRoIjoibW9iaWxlX3BheSIsInBheW1lbnRJbmZvIjp7ImFtb3VudCI6MTAwLCJjdXJyZW5jeSI6IlVTRCJ9LCJwYXltZW50T2JqZWN0IjoibGlua19wYXltZW50X21ldGhvZCIsInBheW1lbnRVc2VyQWdlbnQiOiJ0ZXN0IiwicHVibGlzaGFibGVLZXkiOiJwa190ZXN0XzEyMyJ9")
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Link/LinkURLGeneratorTests.swift:31]: Hmm, based on the code changes you made, i'm not understanding why this URL had to change.

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Utils/LinkURLGenerator.swift [hunk 2]
```diff
@@ -95,7 +95,9 @@ class LinkURLGenerator {
 
 extension LinkURLParams {
     func toURLEncodedBase64() throws -> String {
-        let encodedData = try JSONEncoder().encode(self)
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = [.sortedKeys]
+        let encodedData = try encoder.encode(self)
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Utils/LinkURLGenerator.swift:100]: I'm guessing this gives us repeatability for the link that is being generated?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ## X.X.X
 ### PaymentSheet
+* [Fixed] Fixed an issue with selecting from lists on macOS Catalyst. Note that only macOS 11 or later is supported: We do not recommend releasing a Catalyst app targeting macOS 10.15.
```
Comment (wooj-stripe) [inline CHANGELOG.md:3]: 10.15 (or lower) ?

File: BuildConfigurations/StripeiOS Tests-Debug.xcconfig [hunk 4]
```diff
@@ -12,3 +12,8 @@ GCC_PREPROCESSOR_DEFINITIONS = DEBUG=1 $(inherited)
 DEBUG_INFORMATION_FORMAT = dwarf
 SWIFT_ACTIVE_COMPILATION_CONDITIONS[arch=*] = DEBUG $(inherited)
 SWIFT_OPTIMIZATION_LEVEL = -Onone
+
+// Hack for getting snapshot tests working on Catalyst
+//FRAMEWORK_SEARCH_PATHS[sdk=macosx*] =
```
Comment (wooj-stripe) [inline BuildConfigurations/StripeiOS Tests-Debug.xcconfig:17]: remove commented out `//FRAMEWORK_SEARCH_PATHS[sdk=macosx*] =` ?

File: CHANGELOG.md (hunk 3 above)
Comment (davidme-stripe) [reply CHANGELOG.md:3]: There is no 10.16 (they switched to 11), and there are no Catalyst-supporting macOS versions below 10.15

File: BuildConfigurations/StripeiOS Tests-Debug.xcconfig (hunk 4 above)
Comment (davidme-stripe) [reply BuildConfigurations/StripeiOS Tests-Debug.xcconfig:17]: Yep this is still in progress, trying to get the tests working in CI.

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Utils/LinkURLGenerator.swift (hunk 2 above)
Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Utils/LinkURLGenerator.swift:100]: Yeah, the order of the JSON is different on macOS vs iOS

File: CHANGELOG.md (hunk 3 above)
Comment (wooj-stripe) [reply CHANGELOG.md:3]: Oops. right.

**Review by wooj-stripe — APPROVED** [review]

**Comment by davidme-stripe** [comment]
Bailing on the tests, Catalyst testing is too broken. If we want to bring it up, I had some success with snapshot tests after following https://bugs.webkit.org/attachment.cgi?id=426598&action=diff, linking against /System/Library/PrivateFrameworks/UIKitMacHelper.framework, and calling `UINSApplicationInstantiate()`. But there were then other crashes deep in XCTest, and I haven't had time to figure them out.

**Pull request merged**